test/ziggurat/messaging/producer_test.clj

```diff
@@ -336,3 +436,44 @@
             (is (= "send" (-> finished-spans
                               (.get 0)
                               (.operationName))))))))))
+
+(deftest get-channel-queue-timeout-ms-test
+  (testing "when retries are enabled"
+    (let [topic-entity :default
+          channel :linear-retry
+          message {:foo "bar" :retry-count 2}]
+      (with-redefs [config/ziggurat-config (constantly (assoc (config/ziggurat-config)
+                                                              :stream-router {topic-entity {:channels {channel {:retry {:count 5
+                                                                                                                        :enabled true
+                                                                                                                        :queue-timeout-ms 2000}}}}}))]
+        (is (= 2000 (producer/get-channel-queue-timeout-ms topic-entity channel message))))))
+  (testing "when exponential backoff are enabled and channel retry count not defined"
+    (let [topic-entity :default
+          channel :channel-no-retry-count
+          message {:foo "bar" :retry-count 2}]
+      (with-redefs [config/ziggurat-config (constantly (assoc (config/ziggurat-config)
+                                                              :stream-router {topic-entity {:channels {channel {:retry {:enabled true
```

Comment:
How will this type of channel config work for https://github.com/gojek/ziggurat/blob/master/src/ziggurat/messaging/producer.clj#L96?

There's no :count key here in this config.

Reply:
@mjayprateek if :count key is not exist for specific channel, it will fallback to :ziggurat :retry :count